src/train.jl

```diff
@@ -52,15 +52,16 @@ function setup(rule::Optimisers.AbstractRule, model)
 end
 
 """
-    train!(loss, model, data, opt)
+    train!(loss, model, data, opt; cb=nothing)
 
 Uses a `loss` function and training `data` to improve the `model`'s parameters
 according to a particular optimisation rule `opt`. Iterates through `data` once,
-evaluating `loss(model, d...)` for each `d` in data.
+evaluating `loss(model, d...)` for each `d` in `data` in case of tuple iterates, 
+and `loss(model, d)` otherwise.
```

Comment:
I think this is pretty hard to understand if you don't already know about this weird thing, what's a tuple iterator? Maybe:
```suggestion
evaluating for each `d in data` either `loss(model, d...)` if `d isa Tuple`,
or else   `loss(model, d)` for other `d`.
```